src/newComponents/viewer/LeadPreview/Preview/index.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import {
+    _cs,
+    isNotDefined,
+    isDefined,
+    isValidUrl,
+} from '@togglecorp/fujs';
+import {
+    AiOutlineShrink,
+    AiOutlineExpandAlt,
+} from 'react-icons/ai';
+import {
+    IoOpenOutline,
+} from 'react-icons/io5';
+import {
+    QuickActionButton,
+    QuickActionLink,
+    TextInput,
+    ImagePreview,
+} from '@the-deep/deep-ui';
+
+import {
+    MimeTypes,
+    isImageMimeType,
+    isDocMimeType,
+    isHTMLMimeType,
+    createUrlForGoogleViewer,
+} from './mimeTypes';
+
+import styles from './styles.scss';
+
+function isHttps(text?: string) {
+    if (!text) {
+        return false;
+    }
+    return !!(text || '').match(/^https:\/\//);
+}
+
+interface FinalIframeProps {
+    className?: string;
+    url: string;
+    mimeType: MimeTypes;
+    canShowIframe: boolean;
+}
+
+function FinalIframe(props: FinalIframeProps) {
+    const {
+        className,
+        url,
+        mimeType,
+        canShowIframe,
+    } = props;
+
+    const isRunningInHTTP = window.location.protocol === 'http:';
+    const isCurrentUrlHTTPs = isHttps(url);
+
+    if ((isCurrentUrlHTTPs || isRunningInHTTP) && isImageMimeType(mimeType)) {
```

Comment:
The order of expression would be better if reversed.
Also, comments why we are checking for https here